com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/flowable/FlowableFacade.java

```diff
@@ -101,18 +101,19 @@ private boolean hasHistoricProcessInstanceNotEnded(String processId) {
     }
 
     private boolean hasProcessInstanceNotEnded(String processId) {
-        ProcessInstance processInstance = processEngine.getRuntimeService()
-            .createProcessInstanceQuery()
-            .processInstanceId(processId)
-            .singleResult();
+        ProcessInstance processInstance = getProcessInstance(processId);
         return processInstance != null && !processInstance.isEnded();
     }
 
     private boolean isProcessInstanceDeleted(String processId) {
+        return getProcessInstance(processId) == null && hasDeleteReason(processId);
```

Comment:
This check for whether the process instance is null or not is repeated in a lot of places. Please move it to the beginning of the getProcessInstanceState() method and remove it from the rest of the private methods.